### DISCUSSION TITLE
Helper ask for credentials when KeePassXC database is locked?

### DISCUSSION THREAD

**Discussion by kastl-ars** [Frederick888/git-credential-keepassxc#103] [Q&A] [answered]
Dear Frederick, dear all,

thanks for this nice little helper. Works like a charm.

One thing I am puzzled about: When the KeePassXC database is locked, trying to use credentials results in a login prompt asking for credentials on the command line. So it looks like the helper does not recognize that the database is locked.

```
Dec 18 06:12:38.039 ERRO No valid database associations found in configuration file, Caused by: N/A
```

Can I configure that it ask me to unlock the database instead? That would be very helpful.

Kind Regards
Johannes

P.S.: Merry Christmas!

**Comment by Frederick888** [comment, marked as answer]
`git-credential-keepassxc --help`...

**Reply by kastl-ars** [reply]
Thanks, I missed the `--unlock` argument. Now I am being asked, but after unlocking the database I am still asked for credentials. I'll do some further testing.

This is what I used:

```
[credential]
        helper = "keepassxc --git-groups --unlock 3"
```

**Reply by kastl-ars** [reply]
Huh, and after writing the comment it seems to work flawlessly? Perfect!
